src/main/webapp/lobby.js

```diff
@@ -24,22 +24,41 @@ channel.bind(PUSHER_CHAT_CHANNEL_NAME, function (data) {
 
 // Fetches list of usernames, appends each username to html list.
 function loadUsernames() {
-    fetch('/user').then(response => response.json()).then((users) => {
-      const userList = document.getElementById('user-list');
-      users.forEach((username) => {
-        userList.appendChild(createUsernameElement(username));
-      })
-    });
-  }
+  // Get current url.
+  let url = window.location.href;
+  let paramString = parseRoomId(url);
+
+  fetch(`/room?roomId=${paramString}`).then(response => response.json()).then((users) => {
+    const userList = document.getElementById('user-list');
+    users.forEach((username) => {
+      userList.appendChild(createUsernameElement(username));
+    })
+  });
+}
 
 // Appends text node to list node, returns list node.
 function createUsernameElement(username) {
-    var node = document.createElement("li");
-    var textnode = document.createTextNode(username);
-    node.appendChild(textnode);
-    return node;
+  var node = document.createElement("li");
+  var textnode = document.createTextNode(username);
+  node.appendChild(textnode);
+  return node;
+}
+
+function parseRoomId(url) {
+  // Split url at '?' and saves second half (url parameter).
+  let paramString = url.split('?')[1]; 
+  let queryString = new URLSearchParams(paramString);
+
+  // Saves value into roomId (pair[0] would be key).
+  let roomId = "";
+  for (let pair of queryString.entries()) {
+	  roomId = pair[1]; 
   }
 
-  window.addEventListener('DOMContentLoaded', ()=>{
+  return roomId;
+}
```

Comment:
At some point, this seems like a valid function to add testing too

Reply:
@salnad yes I added some yesterday

Reply:
@salnad yes I added some yesterday in the "Add joining user to room" PR